tests/components/weather/test_init.py

```diff
@@ -191,3 +184,225 @@ async def test_none_forecast(
     assert forecast[ATTR_FORECAST_PRESSURE] is None
     assert forecast[ATTR_FORECAST_WIND_SPEED] is None
     assert forecast[ATTR_FORECAST_PRECIPITATION] is None
+
+
+async def test_custom_units(hass: HomeAssistant, enable_custom_integrations) -> None:
+    """Test custom unit."""
+    wind_speed_value = 5
+    wind_speed_unit = SPEED_METERS_PER_SECOND
+    pressure_value = 110
+    pressure_unit = PRESSURE_HPA
+    temperature_value = 20
+    temperature_unit = TEMP_CELSIUS
+    visibility_value = 11
+    visibility_unit = LENGTH_KILOMETERS
+
+    set_options = {
+        "wind_speed_unit_of_measurement": SPEED_MILES_PER_HOUR,
+        "precipitation_unit_of_measurement": LENGTH_INCHES,
```

Comment:
This doesn't work because the entity has no precipitation unit.
Also, the precipitation is not checked below.